streams/src/main/java/org/apache/kafka/streams/state/internals/InMemoryTimeOrderedKeyValueBuffer.java

```diff
@@ -375,42 +341,61 @@ private void restoreBatch(final Collection<ConsumerRecord<byte[], byte[]>> batch
         updateBufferMetrics();
     }
 
+    private byte[] inferPriorValue(final Bytes key, final byte[] serializedChange) {
+        final byte[] priorValue;
+        if (index.containsKey(key)) {
+            priorValue = internalPriorValueForBuffered(key);
+        } else {
+            final Change<V> change = valueSerde.deserializer().deserialize(
```

Comment:
Similarly here, we deserialized the whole `Changed` object but we only need the old-value here, since I know exactly how `ChangedSerde` works can we skip deserializing the new-value?

Reply:
Good call here as well. I added a utility method to the FullChangeSerde for this.